File: src/main/java/com/refinedmods/refinedstorage/container/slot/DisabledSlot.java
package com.refinedmods.refinedstorage.container.slot;

import net.minecraft.world.item.ItemStack;
import net.neoforged.neoforge.items.IItemHandler;
import javax.annotation.Nonnull;

public class DisabledSlot extends BaseSlot {
    public DisabledSlot(IItemHandler itemHandler, int inventoryIndex, int x, int y) {
        super(itemHandler, inventoryIndex, x, y);
    }

    @Override
    public boolean mayPlace(@Nonnull ItemStack stack) {
        return false;
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/slot/DisabledSlot.kt
package com.refinedmods.refinedstorage.container.slot

import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import net.minecraft.item.ItemStack
import net.minecraftforge.items.IItemHandler

class DisabledSlot(itemHandler: IItemHandler?, inventoryIndex: Int, x: Int, y: Int) : BaseSlot(itemHandler, inventoryIndex, x, y) {
    fun isItemValid(@Nonnull stack: ItemStack?): Boolean {
        return false
    }
}